apps/ui/lib/lens/misc/Flowdock/Messages.tsx

```diff
@@ -0,0 +1,181 @@
+import { circularDeepEqual } from 'fast-equals';
+import styled from 'styled-components';
+import React, { useEffect, useRef, useState } from 'react';
+import { Flex, Box, Txt } from 'rendition';
+import { Column, Icon } from '../../../components';
+import _ from 'lodash';
+import { GroupedVirtuoso } from 'react-virtuoso';
+import { SdkQueryOptions } from '@balena/jellyfish-client-sdk/build/types';
+import { JsonSchema } from 'autumndb';
+import { useCursorEffect } from '../../../hooks';
+
+import { Markdown } from 'rendition/dist/extra/Markdown';
+
+const MessageListColumn = styled(Column)`
+	position: relative;
+	min-height: 0;
+`;
+
+const MessageContainer = styled(Box)`
+	min-width: 0;
+	border-radius: 12px;
+	border-top-left-radius: 0;
+	box-shadow: -5px 4.5px 10.5px 0 rgba(152, 173, 227, 0.08);
+	a {
+		color: inherit;
+		text-decoration: underline;
+	}
+
+	img {
+		background-color: transparent !important;
+		&.emoji {
+			width: 20px;
+			height: 20px;
+			vertical-align: middle;
+		}
+	}
+	code {
+		color: #333;
+		background-color: #f6f8fa;
+	}
+
+	border: solid 0.5px #e8ebf2;
+	background: white;
+`;
+
+const HeaderWrapper = styled(Flex)`
+	position: relative;
+`;
+
+const getQuery = (threadId: string): JsonSchema => {
+	return {
+		type: 'object',
+		required: ['type'],
+		properties: {
+			type: {
+				type: 'string',
+				const: 'flowdock-message@1.0.0',
+			},
+		},
+		$$links: {
+			'is attached to': {
+				type: 'object',
+				required: ['type', 'id'],
+				properties: {
+					type: {
+						type: 'string',
+						const: 'flowdock-archive@1.0.0',
+					},
+					id: {
+						type: 'string',
+						const: threadId,
+					},
+				},
+			},
+		},
+	};
+};
+
+const INITIAL_ITEM_COUNT = 30;
+
+const START_INDEX = 10000;
+
+const DEFAULT_OPTIONS: SdkQueryOptions = {
+	limit: 30,
+	sortBy: 'created_at',
```

Comment:
I want to sort by 'data.sent' field in `flowdock-message` card. but the server is throwing errors. tried various things. couldn't pin point on the issue.